### PR TITLE
Docs/doc improvements 3

### PR DESCRIPTION
**Pull request by jvr0x**
This PR further improves the documentation, addressing the **get started** section.

### Checklist

- [ ] CI is green.
- [ ] Changelog updated.

### CODE REVIEW

**Comment by CyberHoward** [comment]
Awesome @jvr0x! Here's some feedback going through the docs:

- I think the first page is a bit bare. Maybe some images would flare it up a bit? I'd like there to be an overview of the components of Abstract, i.e. what we're working on. 
Example: 
image. 

- Some of the links don't work.
- The use-cases section would be more valuable if actual examples are given with a bit more detail (like 4T2, a DCA app, subscriptions with contributors (upcoming)). 
- Code snippet on `Account Abstraction` is broken
- on [this page](https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/1_abstract_sdk) clicking the next button is broken so there's probably something wrong with the routing.
- Bottom of [account abstraction](https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/3_account_abstraction) mentions exploring the Account architecture on the next section but the next section is on SDK and cw-orch. Only then is the account architecture. 
- cw-orch page is in sidebar but doesn't exist.
- In [this page)(https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/4_architecture) "Let’s see how the account architecture achieves this.
Let’s dive deeper into the two components of the Abstract Account."
- (nit) On [this page](https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/7_module_types) apps are mentioned first in the module type list but adapters are explained first.
- On [this](https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/7_module_types) page. What does the image in the app section represent? What are the arrows? How does this image help the reader understand what apps are?
- (nit) On the same page, the steps could be **bold** instead of headings. Also, the relevant modules listed are not really a thing. Maybe you can just link to the modules directory in our monorepo?. 
- On the same page, the non-dependent execution can use some text. In general I'd like to see some more explanation on what's different between these three modes. Maybe the naming of the modes can be revised as well. Happy to jump on a short call to ideate. 
- [this page](https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/8_upgradability) mentions API models. We renamed API to adapter. 
- On the same page section [Upgrading Modules](https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/8_upgradability#upgrading-modules) it would be nice to add some more detail on how the integrity is ensured. I.e. how the proposed module upgrades are performed sequentially while keeping track of the changes and doing a version-requirement check at the end of all the migrations. The diagram explains this well but a little bit of texts would make it easier to understand. 
- (nit) The text on [here](https://docs-doc-improvements-3.abstract-docs.pages.dev/5_platform/) could use some work.
- [The ans page](https://docs-doc-improvements-3.abstract-docs.pages.dev/5_platform/ans) doesn't include the pool storage. 
- The same page still uses `memory contract`, which should be `ANS contract`.
- The first mentioning of monetization is [the monetization page](https://docs-doc-improvements-3.abstract-docs.pages.dev/5_platform/monetization). IMO this should be mentioned earlier as one of the unique aspects of the platform. 
- On the same page you can mention that subscriptions are being worked on! 

Will do the rest in a bit :)

**Comment by jvr0x** [comment]
> Awesome @jvr0x! Here's some feedback going through the docs:
> 
>     * I think the first page is a bit bare. Maybe some images would flare it up a bit? I'd like there to be an overview of the components of Abstract, i.e. what we're working on.
>       Example:
>       image.
> 
>     * Some of the links don't work.
> 
>     * The use-cases section would be more valuable if actual examples are given with a bit more detail (like 4T2, a DCA app, subscriptions with contributors (upcoming)).
> 
>     * Code snippet on `Account Abstraction` is broken
> 
>     * on [this page](https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/1_abstract_sdk) clicking the next button is broken so there's probably something wrong with the routing.
> 
>     * Bottom of [account abstraction](https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/3_account_abstraction) mentions exploring the Account architecture on the next section but the next section is on SDK and cw-orch. Only then is the account architecture.
> 
>     * cw-orch page is in sidebar but doesn't exist.
> 
>     * In [this page)(https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/4_architecture) "Let’s see how the account architecture achieves this.
>       Let’s dive deeper into the two components of the Abstract Account."
> 
>     * (nit) On [this page](https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/7_module_types) apps are mentioned first in the module type list but adapters are explained first.
> 
>     * On [this](https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/7_module_types) page. What does the image in the app section represent? What are the arrows? How does this image help the reader understand what apps are?
> 
>     * (nit) On the same page, the steps could be **bold** instead of headings. Also, the relevant modules listed are not really a thing. Maybe you can just link to the modules directory in our monorepo?.
> 
>     * On the same page, the non-dependent execution can use some text. In general I'd like to see some more explanation on what's different between these three modes. Maybe the naming of the modes can be revised as well. Happy to jump on a short call to ideate.
> 
>     * [this page](https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/8_upgradability) mentions API models. We renamed API to adapter.
> 
>     * On the same page section [Upgrading Modules](https://docs-doc-improvements-3.abstract-docs.pages.dev/3_framework/8_upgradability#upgrading-modules) it would be nice to add some more detail on how the integrity is ensured. I.e. how the proposed module upgrades are performed sequentially while keeping track of the changes and doing a version-requirement check at the end of all the migrations. The diagram explains this well but a little bit of texts would make it easier to understand.
> 
>     * (nit) The text on [here](https://docs-doc-improvements-3.abstract-docs.pages.dev/5_platform/) could use some work.
> 
>     * [The ans page](https://docs-doc-improvements-3.abstract-docs.pages.dev/5_platform/ans) doesn't include the pool storage.
> 
>     * The same page still uses `memory contract`, which should be `ANS contract`.
> 
>     * The first mentioning of monetization is [the monetization page](https://docs-doc-improvements-3.abstract-docs.pages.dev/5_platform/monetization). IMO this should be mentioned earlier as one of the unique aspects of the platform.
> 
>     * On the same page you can mention that subscriptions are being worked on!
> 
> 
> Will do the rest in a bit :)

Massive feedback, thanks for pointhing those out, will revise!

File: framework/docs/src/3_framework/1_abstract_sdk.md [hunk 1]
```diff
@@ -75,11 +75,11 @@ types of bases available, each tailored for specific needs and functionalities.
 
 There are three kinds of bases:
 
-- [App]((https://crates.io/crates/abstract-app){target="_blank"})
-- [Adapter]((https://crates.io/crates/abstract-adapter){target="_blank"})
-- [IBC-host]((https://crates.io/crates/abstract-ibc-host){target="_blank"})
+- <a href="https://crates.io/crates/abstract-app" target="_blank">App</a>
+- <a href="https://crates.io/crates/abstract-adapter" target="_blank">Adapter</a>
+- <a href="https://crates.io/crates/abstract-ibc-host" target="_blank">IBC-host</a>
```
Comment (CyberHoward) [inline framework/docs/src/3_framework/1_abstract_sdk.md:80]: This base is not supported anymore

**Review by CyberHoward — CHANGES_REQUESTED** [review]
Great revision, thanks a lot!! 

Some of the code-snippets also don't seem to render for me. The specified paths are relative paths so it's important where the doc-serve command is ran. How are you serving it yourself? 

I'm using the `just docs-serve` command in the justfile. The miss-renders are in 

http://localhost:3000/3_framework/2_account_abstraction.html
http://localhost:3000/4_get_started/4_sdk.html
http://localhost:3000/4_get_started/6_module_testing.html

Also what do you think about maybe merging/restructuring the first three pages? I'm having a hard time identifying what's conceptually different about them and it seems like they share a lot of content.

**Review by CyberHoward — APPROVED** [review]

**Pull request merged**